plugins/modules/net_tools/dnsimple_info.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#  Copyright: Edward Hilgendorf, <edward@hilgendorf.me>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: dnsimple_info
+
+short_description: Pull basic info from DNSimple API.
+
+version_added: "4.1.0"
+
+description: Retrieve existing records and domains from DNSimple API.
+
+options:
+    name:
+        description:
+          - The domain name to retrieve info from.
+          - Will return all associated records for this domain if specified.
+          - If not specified, will return all domains associated with the account ID.
+        type: str
+
+    account_id:
+        description: The account ID to query.
+        required: true
+        type: str
+
+    api_key:
+        description: The API key to use.
+        required: true
+        type: str
+
+    record:
+        description:
+          - The record to find.
+          - If specified, only this record will be returned instead of all records.
+        required: false
+        type: str
+
+    sandbox:
+        description: Whether or not to use sandbox environment.
+        required: false
+        default: false
+        type: bool
+
+author:
+    -  Edward Hilgendorf (@edhilgendorf)
+'''
+
+EXAMPLES = r'''
+- name: Get all domains from an account
+  community.general.dnsimple_info:
+    account_id: "1234"
+    api_key: "1234"
+
+- name: Get all records from a domain
+  dnsimple_info:
+    name: "example.com"
+    account_id: "1234"
+    api_key: "1234"
+
+- name: Get all info from a matching record
+  dnsimple_info:
+    name: "example.com"
+    record: "subdomain"
+    account_id: "1234"
+    api_key: "1234"
+'''
+
+RETURN = r'''
+dnsimple_domain_info:
+    description: Returns a list of dictionaries of all domains associated with the supplied account ID.
+    type: list
+    elements: dict
+    returned: success
+    sample:
+    - account_id: 1234
+      created_at: '2021-10-16T21:25:42Z'
+      id: 123456
+      last_transferred_at:
+      name: example.com
+      reverse: false
+      secondary: false
+      updated_at: '2021-11-10T20:22:50Z'
```

Comment:
It's probably also useful to the user if you document the fields that are returned.
```suggestion
      updated_at: '2021-11-10T20:22:50Z'
    contains:
      account_id:
        description: The account ID.
        type: int
      created_at:
        description: When the domain entry was created.
        type: str
      ...
```